### PR TITLE
Fix VR inheriting boot settings from user's VM

### PR DESCRIPTION
**Pull request by hsato03**
### Description

In a situation where there is a stopped user VM and this VM's network VR is also stopped, starting the user VM will also cause the VR to start. However, in this situation, the VR is inheriting the user VM's boot parameters, causing it to not start when the user VM has the UEFI boot type.

Because of this, the inheritance of user VM boot parameters has been removed from VR and other boot parameters have been defined in their place.

Fixes: #8734

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?

I followed the same steps from `How did you try to break this feature and the system with this change?` and verified that the user's VM and VR started correctly.

#### How did you try to break this feature and the system with this change?

1. Using VMware, I created an isolated network (`isolated`);
2. I deployed a VM (`test`) using the `isolated` network and started it;
3. After implementing the `isolated` network, I stopped the VM `test` and the VR of the `isolated` network;
4. I added the `UEFI=SECURE` setting to the `test` VM and started it again, causing the `isolated` network VR to start as well;
5. The VR could not start because it had the UEFI boot type.

### CODE REVIEW

**Comment by hsato03** [comment]
I will keep it in draft because I'm still testing it, but it's ready to be reviewed.

**Review by DaanHoogland — COMMENTED** [review]
code looks good but I wonder if ReProgramGuestNetworks is really the only parameter that needs to be set.

File: server/src/main/java/com/cloud/network/element/VirtualRouterElement.java [hunk 1]
```diff
@@ -272,12 +272,15 @@ public boolean prepare(final Network network, final NicProfile nic, final Virtua
             return false;
         }
 
+        Map<VirtualMachineProfile.Param, Object> params = new HashMap<>();
```
Comment (weizhouapache) [inline server/src/main/java/com/cloud/network/element/VirtualRouterElement.java:275]:
```suggestion
        implement(network, offering, dest, context);
```

Comment (weizhouapache) [reply server/src/main/java/com/cloud/network/element/VirtualRouterElement.java:275]: @DaanHoogland 
right. all lines before `return true`

github did not work well when I selected many lines. .

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/network/element/VirtualRouterElement.java:275]: i know, to bad, it would help with suggestions like these.

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test ubuntu22 kvm-ubuntu22

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Pull request merged**